chiselFrontend/src/main/scala/chisel3/core/Aggregate.scala

```diff
@@ -43,12 +43,16 @@ object Vec {
     *
     * @note elements are NOT assigned by default and have no value
     */
-  def apply[T <: Data](n: Int, gen: T)(implicit compileOptions: CompileOptions): Vec[T] =
-    new Vec(gen.chiselCloneType, n)
+  def apply[T <: Data](n: Int, gen: T): Vec[T] = macro VecTransform.apply_ngen;
+
+  def do_apply[T <: Data](n: Int, gen: T)(implicit sourceInfo: SourceInfo, compileOptions: CompileOptions): Vec[T] = {
+    if ( gen.isLit ) Vec( Seq.fill(n)(gen) )
+    else new Vec(gen.chiselCloneType, n)
+  }
 
   @deprecated("Vec argument order should be size, t; this will be removed by the official release", "chisel3")
-  def apply[T <: Data](gen: T, n: Int)(implicit compileOptions: CompileOptions): Vec[T] =
-    apply(n, gen)
+  def apply[T <: Data](gen: T, n: Int)(implicit sourceInfo: SourceInfo, compileOptions: CompileOptions): Vec[T] =
```

Comment:
For deprecated methods I usually use UnlocatableSourceInfo / DeprecatedSourceInfo, because it's not worth the pain of threading source info through